apps/dashboard/src/components/workflow-editor/steps/email/html-editor.tsx

```diff
@@ -102,24 +112,73 @@ export function HtmlEditor({ value, onChange }: HtmlEditorProps) {
     ];
   }, []);
 
+  const handleFormatClick = useCallback(
+    async (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation();
+      e.preventDefault();
+
+      const formattedValue = await format(value, {
+        parser: 'liquid-html',
+        printWidth: 120,
+        tabWidth: 2,
+        useTabs: false,
+        htmlWhitespaceSensitivity: 'css',
+        plugins: [parserHtml, parserLiquid],
+      });
+
+      onChange(formattedValue);
+      saveForm();
+    },
+    [value, onChange, saveForm]
+  );
+
+  const handleEditorBlur = useCallback((e: React.FocusEvent<HTMLDivElement, Element>) => {
+    // if the blur happens on the format button, we don't want to trigger blur on the editor
+    // because it will save the form unformatted and than format it again
+    if (e.relatedTarget === formatButtonRef.current) {
+      e.stopPropagation();
+      e.preventDefault();
+      return;
+    }
+  }, []);
```

Comment:
when blur happens on the editor and the relatedTarget is the format button then prevent the blur